lib/datadog/core/utils/only_once.rb

```diff
@@ -31,9 +31,7 @@ def ran?
           @mutex.synchronize { @ran_once }
         end
 
-        private
-
-        def reset_ran_once_state_for_tests
+        def reset
           @mutex.synchronize { @ran_once = false }
         end
```

Comment:
I don't think this is the right way to go -- I think this is bending the `OnlyOnce` in a really weird direction generally only to support a specific use-case of one of its users.

Rather than resetting the only once, consider creating a new instance?